pkg/controller/user-cluster-controller-manager/resources/controller.go

```diff
@@ -305,6 +307,7 @@ type reconciler struct {
 	ccmMigration                 bool
 	ccmMigrationCompleted        bool
 	enableOperatingSystemManager bool
+	kubernetesDashboardEnabled   bool
```

Comment:
Do you actually need this? Your code is already correctly using cluster.Spec everywhere in the reconciler. Ideally the usercluster-ctrl-mgr should react dynamically to changes in the cluster without needing to restart.

Reply:
`usercluster-ctrl-mgr` depends on the flags and not on the cluster.Spec. Maybe I'm misunderstanding your comment?

Reply:
I've pushed a commit into your branch to show what I am talking about. I see where you came from and what you took as a template, but I think we can do better than the old code that relied on CLI flags. :)

Let me know what you think

Reply:
Oh, so you meant that, I was also curious why we don't do this already. I love the idea and it sounds super good to me 👍